pkg/operator/ceph/cluster/cluster.go

```diff
@@ -261,78 +261,18 @@ func (c *cluster) createInitialCrushMap() error {
 }
 
 func clusterChanged(oldCluster, newCluster cephv1.ClusterSpec, clusterRef *cluster) bool {
-	changeFound := false
-	oldStorage := oldCluster.Storage
-	newStorage := newCluster.Storage
 
 	// sort the nodes by name then compare to see if there are changes
-	sort.Sort(rookv1alpha2.NodesByName(oldStorage.Nodes))
-	sort.Sort(rookv1alpha2.NodesByName(newStorage.Nodes))
-	if !reflect.DeepEqual(oldStorage.Nodes, newStorage.Nodes) {
-		logger.Infof("The list of nodes has changed")
-		changeFound = true
-	}
-
-	if oldCluster.Dashboard.Enabled != newCluster.Dashboard.Enabled {
-		logger.Infof("dashboard enabled has changed from %t to %t", oldCluster.Dashboard.Enabled, newCluster.Dashboard.Enabled)
-		changeFound = true
-	}
-	if oldCluster.Dashboard.UrlPrefix != newCluster.Dashboard.UrlPrefix {
-		logger.Infof("dashboard url prefix has changed from \"%s\" to \"%s\"", oldCluster.Dashboard.UrlPrefix, newCluster.Dashboard.UrlPrefix)
-		changeFound = true
-	}
-
-	if oldCluster.Dashboard.Port != newCluster.Dashboard.Port {
-		logger.Infof("dashboard port has changed from \"%d\" to \"%d\"", oldCluster.Dashboard.Port, newCluster.Dashboard.Port)
-		changeFound = true
-	}
-
-	if (oldCluster.Dashboard.SSL == nil && newCluster.Dashboard.SSL != nil) ||
-		(oldCluster.Dashboard.SSL != nil && newCluster.Dashboard.SSL == nil) ||
-		(oldCluster.Dashboard.SSL != nil && newCluster.Dashboard.SSL != nil &&
-			*oldCluster.Dashboard.SSL != *newCluster.Dashboard.SSL) {
-		oldSSL := "<default>"
-		if oldCluster.Dashboard.SSL != nil {
-			oldSSL = strconv.FormatBool(*oldCluster.Dashboard.SSL)
-		}
-		newSSL := "<default>"
-		if newCluster.Dashboard.SSL != nil {
-			newSSL = strconv.FormatBool(*newCluster.Dashboard.SSL)
-		}
-		logger.Infof("dashboard ssl option has changed from \"%s\" to \"%s\"", oldSSL, newSSL)
-		changeFound = true
-	}
+	sort.Sort(rookv1alpha2.NodesByName(oldCluster.Storage.Nodes))
+	sort.Sort(rookv1alpha2.NodesByName(newCluster.Storage.Nodes))
 
-	if oldCluster.Mon.Count != newCluster.Mon.Count {
-		logger.Infof("number of mons have changed from %d to %d. The health check will update the mons...", oldCluster.Mon.Count, newCluster.Mon.Count)
-		clusterRef.mons.MonCountMutex.Lock()
-		clusterRef.mons.Count = newCluster.Mon.Count
-		clusterRef.mons.MonCountMutex.Unlock()
+	// any change in the crd will trigger an orchestration
+	if !reflect.DeepEqual(oldCluster, newCluster) {
```

Comment:
For my own clarity, this `DeepEqual` is part of the generated code, yes?

Reply:
it's actually on the `reflect` package. So if any fields change in the struct, we will notice the difference. Maybe you're thinking of the `DeepCopy()` on the generated code, I don't see a `DeepEqual()` there